tests/unit_tests/export_detection_model_test.py

```diff
@@ -0,0 +1,713 @@
+            export_result = ppyolo_e.export(
+                out_path,
+                device=device,
+                engine=ExportTargetBackend.TENSORRT,
+                max_predictions_per_image=max_predictions_per_image,
+                input_image_shape=(640, 640),
+                output_predictions_format=DetectionOutputFormatMode.BATCH_FORMAT,
+                quantization_mode=ExportQuantizationMode.FP16,
+            )
+            assert export_result is not None
+
+    def test_export_with_int8_quantization(self):
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            out_path = os.path.join(tmpdirname, "ppyoloe_s_with_int8_quantization.onnx")
+
+            dummy_calibration_dataset = [torch.randn((3, 640, 640), dtype=torch.float32) for _ in range(32)]
+            dummy_calibration_loader = DataLoader(dummy_calibration_dataset, batch_size=8, num_workers=0)
+
+            ppyolo_e: ExportableObjectDetectionModel = models.get(Models.PP_YOLOE_S, pretrained_weights="coco")
+            export_result = ppyolo_e.export(
+                out_path,
+                engine=ExportTargetBackend.ONNXRUNTIME,
+                max_predictions_per_image=300,
+                input_image_shape=(640, 640),
+                output_predictions_format=DetectionOutputFormatMode.BATCH_FORMAT,
+                quantization_mode=ExportQuantizationMode.INT8,
+                calibration_loader=dummy_calibration_loader,
+            )
+
+            num_predictions, pred_boxes, pred_scores, pred_classes = self._run_inference_with_onnx(export_result)
+
+            assert num_predictions.shape == (1, 1)
+            assert pred_boxes.shape == (1, 300, 4)
+            assert pred_scores.shape == (1, 300)
+            assert pred_classes.shape == (1, 300)
+            assert pred_classes.dtype == np.int64
+
+    def test_export_quantized_with_calibration_to_tensorrt(self):
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            out_path = os.path.join(tmpdirname, "pp_yoloe_s_quantized_with_calibration.onnx")
+
+            dummy_calibration_dataset = [torch.randn((3, 640, 640), dtype=torch.float32) for _ in range(32)]
+            dummy_calibration_loader = DataLoader(dummy_calibration_dataset, batch_size=8)
+
+            ppyolo_e: ExportableObjectDetectionModel = models.get(Models.PP_YOLOE_S, pretrained_weights="coco")
+            export_result = ppyolo_e.export(
+                out_path,
+                engine=ExportTargetBackend.TENSORRT,
+                max_predictions_per_image=300,
+                input_image_shape=(640, 640),
+                output_predictions_format=DetectionOutputFormatMode.BATCH_FORMAT,
+                quantization_mode=ExportQuantizationMode.INT8,
+                calibration_loader=dummy_calibration_loader,
+            )
+            assert export_result is not None
+
+    def test_export_yolonas_quantized_with_calibration_to_tensorrt(self):
```

Comment:
Is this supposed to work on a machine without gpu ?
Take in account that the unit tests suite does not run on spots....

Reply:
Yep, it runs fine - exporting to TRT does not involve TRT itself. We still export through ONNX